lib/neo4j-server/cypher_session.rb

```diff
@@ -23,6 +23,10 @@ def initialize(data_url, connection)
       # @return [Faraday]
       # @see https://github.com/lostisland/faraday
       def self.create_connection(params, url = nil)
+        faraday_options = params.delete(:faraday_options) || params.delete('faraday_options') || {}
+        adapter = (faraday_options[:adapter] || faraday_options['adapter'] || :net_http_persistent).to_sym
```

Comment:
OK, it was a bit tricky since it's not just a straight passthrough but I found some time to get that in as well.